frontend/src/features.ts

```diff
@@ -25,9 +25,18 @@ export function initFeatures() {
     window.__FEATURE_FLAGS__ = JSON.stringify(features);
     return;
   }
-  if (!localStorage.getItem('flags')) {
-    localStorage.setItem('flags', JSON.stringify(features));
+  if (localStorage.getItem('flags')) {
+    const originalFlags = localStorage.getItem('flags');
+    const originalFlagsJSON = JSON.parse(originalFlags!);
```

Comment:
Can you handle the exception when JSON.parse is not successful? There is no guarantee that the originalFlags content is valid.

```
    let originalFlagsJSON = []
    try {
        originalFlagsJSON = JSON.parse(originalFlags!);
        ... setting flag ...
    } catch(e) {
        console.warn('Original feature flags format is null or not recognizable, overwriting with default feature flags.');
    }
```